src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -1501,6 +1501,62 @@ private void CheckRecordMemberNames(DiagnosticBag diagnostics)
             {
                 diagnostics.Add(ErrorCode.ERR_CloneDisallowedInRecord, member.Locations[0]);
             }
+
+            var targetEqualsMethod = new SignatureOnlyMethodSymbol(
```

Comment:
This could probably be simplified by moving the check into `AddSynthesizedRecordMembersIfNecessary`. We can have the helper local functions return a boolean to signify whether the method was synthesized or user-defined, then we can just check a couple of such values.

Reply:
@jcouv I didn't get how this can exactly be done. But after giving another look, I actually wonder if this can be done be some tweaks around:

https://github.com/dotnet/roslyn/blob/0d6a84f32345fdd9dbeb7a6e585187b6320194e6/src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs#L2103-L2108.

Below this check, warnings are reported for classes. Why not remove the record condition and try to tweak things below it?
Why even this check existed in the first place?

I also want to ask if this should check for  `object.Equals`, `IEquatable<T>.Equals`, and `GetHashCode`? or only `IEquatable<T>.Equals`, and `GetHashCode` (which is what I've currently)?

Reply:
It feels like there's no need to check overrides of `object.Equals` on records, because it's disallowed to explicitly override that in a record. (hopefully I'm interpreting the question correctly?)

Reply:
@RikkiGibson Thanks! I didn't know it's disallowed. This just came to mind when I saw the checks that applies for classes.

Reply:
@Youssef1313 Here's more details on what I was proposing above.
Instead of search for user-defined methods again, we modify `addGetHashCode` to return the method, then we can just do:
`if ((thisEquals is not SynthesizedRecordEquals) ^ (getHashCode is not SynthesizedRecordEquals)) { /* report diagnostic /* }`